src/store/types.ts

```diff
@@ -0,0 +1,59 @@
+export const SET_USER = 'SET_USER';
+export const SIGN_OUT = 'SIGN_OUT';
+export const SET_LOADING = 'SET_LOADING';
+export const SET_ERROR = 'SET_ERROR';
+export const SET_SUCCESS = 'SET_SUCCESS';
+
```

Comment:
Should be able to create hierarchy these. I am seeing duplicated `password` fields and others.

Use `&` to concatenate two types.

```typescript
type A = {};
type B = A & B;
```

is equivalent to
```typescript
interface A {}
interface B extends A {}
```

Reply:
Something you said that you were going to look into.